include/okapi/impl/device/opticalSensor.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+enum class OpticalSensorOutput {
+  hue,        ///< The color.
+  saturation, ///< The color's intensity relative to its brightness.
+  brightness  ///< The amount of light.
+};
+
+class OpticalSensor : public ControllerInput<double> {
+  public:
+  /**
+   * An optical sensor on a V5 port.
+   *
+   * ```cpp
+   * auto osHue = OpticalSensor(1);
+   * auto osSat = OpticalSensor(1, OpticalSensorOutput::saturation);
+   * ```
+   *
+   * @param iport The V5 port the device uses.
+   * @param ioutput Which sensor output to return from (@ref okapi::OpticalSensor::get).
+   * @param ifilter The filter to use to filter the sensor output. Only the selected output (via
+   * ``ioutput``) is filtered; the other outputs are untouched.
+   */
+  OpticalSensor(std::uint8_t iport,
+                OpticalSensorOutput ioutput = OpticalSensorOutput::hue,
+                std::unique_ptr<Filter> ifilter = std::make_unique<PassthroughFilter>());
+
+  virtual ~OpticalSensor();
```

Comment:
Should be defaulted here instead of in the cpp file.

Reply:
Why?

Reply:
Because they just have `DistanceSensor::~DistanceSensor() = default;` in the source file.

Reply:
Oh oops I confused the class. Maybe the optical sensor has a non default destructor, in that case it's fine.